carapace-server/src/main/java/org/carapaceproxy/server/HttpProxyServer.java

```diff
@@ -156,22 +227,42 @@ public void startAdminInterface() throws Exception {
         context.addServlet(jerseyServlet, "/api/*");
         context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
         context.setAttribute("server", this);
-        contexts.addHandler(context);
 
-        File webUi = new File(basePath, "web/ui");
-        if (webUi.isDirectory()) {
-            WebAppContext webApp = new WebAppContext(webUi.getAbsolutePath(), "/ui");
-            contexts.addHandler(webApp);
-        } else {
-            LOG.severe("Cannot find " + webUi.getAbsolutePath() + " directory. Web UI will not be deployed");
-        }
+        NCSARequestLog requestLog = new NCSARequestLog();
+        requestLog.setFilename(adminAccessLogPath);
+        requestLog.setFilenameDateFormat("yyyy-MM-dd");
+        requestLog.setRetainDays(adminLogRetentionDays);
+        requestLog.setAppend(true);
+        requestLog.setExtended(true);
+        requestLog.setLogCookies(false);
+        requestLog.setLogTimeZone(adminAccessLogTimezone);
+        RequestLogHandler requestLogHandler = new RequestLogHandler();
+        requestLogHandler.setRequestLog(requestLog);
+        requestLogHandler.setHandler(context);
+
+        contexts.addHandler(requestLogHandler);
 
         adminserver.start();
-        String apiUrl = "http://" + adminServerHost + ":" + adminServerPort + "/api";
-        String uiUrl = "http://" + adminServerHost + ":" + adminServerPort + "/ui";
-        metricsUrl = "http://" + adminServerHost + ":" + adminServerPort + "/metrics";
-        LOG.info("Base Admin UI url: " + uiUrl);
-        LOG.info("Base Admin/API url: " + apiUrl);
+
+        LOG.info("Admin UI started");
+
+        if (adminServerHttpPort == 0 && httpConnector != null) {
+            adminServerHttpPort = httpConnector.getLocalPort();
+        }
+        if (adminServerHttpsPort == 0 && httpsConnector != null) {
+            adminServerHttpsPort = httpsConnector.getLocalPort();
+        }
+
+        if (adminServerHttpPort > 0) {
+            LOG.info("Base HTTP Admin UI url: http://" + adminServerHost + ":" + adminServerHttpPort + "/ui");
+            LOG.info("Base HTTP Admin API url: http://" + adminServerHost + ":" + adminServerHttpPort + "/api");
+        }
+        if (adminServerHttpsPort > 0) {
+            LOG.info("Base HTTPS Admin UI url: https://" + adminServerHost + ":" + adminServerHttpsPort + "/ui");
+            LOG.info("Base HTTPS Admin API url: https://" + adminServerHost + ":" + adminServerHttpsPort + "/api");
+        }
+
+        metricsUrl = "http://" + adminServerHost + ":" + adminServerHttpPort + "/metrics";
```

Comment:
This can be wrong if you have only https.

Use http if available, then fallback to https